### PR TITLE
Add Core library.

### PR DESCRIPTION
**Pull request by mamokarz**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -48,6 +48,7 @@ target_include_directories(azure_ulib_c
     PRIVATE
         ${PROJECT_SOURCE_DIR}/deps/umock-c/inc
         ${PROJECT_SOURCE_DIR}/deps/azure-macro-utils-c/inc
+        ${PROJECT_SOURCE_DIR}/deps/azure-core-c/inc
```
Comment (danewalton) [inline CMakeLists.txt:51]: this should not be needed. this should be taken care of with `target_link_libraries` below

File: CMakeLists.txt [hunk 2]
```diff
@@ -69,12 +70,18 @@ if(${remove_ipc_unpublish})
...
+    az::core
+)
+
 set(AZURE_ULIB_C_INC_FOLDER ${CMAKE_CURRENT_LIST_DIR}/inc CACHE INTERNAL "this is what needs to be included if using sharedLib lib" FORCE)
 
 add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/deps/azure-macro-utils-c EXCLUDE_FROM_ALL)
 add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/deps/umock-c EXCLUDE_FROM_ALL)
 add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/deps/azure-ctest EXCLUDE_FROM_ALL)
 add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/deps/azure-c-testrunnerswitcher EXCLUDE_FROM_ALL)
+add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/deps/azure-core-c EXCLUDE_FROM_ALL)
```
Comment (danewalton) [inline CMakeLists.txt:84]: `EXCLUDE_FROM_ALL` can be removed from these

**Review by smcd253 — APPROVED** [review]
looks good to me. should wait until pipeline with updated cmake runs build before merging.

**Review by danewalton — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (mamokarz) [reply CMakeLists.txt:51]: Interesting. 

---
In reply to: [564928784](https://github.com/Azure/azure-ulib-c/pull/57#discussion_r564928784) [](ancestors = 564928784)

**Comment by mamokarz** [comment]
> cmake_minimum_required(VERSION 3.2.0)

Update all to fit 3.10

---
Refers to: CMakeLists.txt:4 in c6f27a6. [](commit_id = c6f27a69173408f05f536e9dc49a68f3e256f2b2, deletion_comment = False)

**Comment by mamokarz** [comment]
> cmake_minimum_required(VERSION 3.2.0)

Thinking better about it, I'll keep version 3.2 for now.

---
In reply to: [767928447](https://github.com/Azure/azure-ulib-c/pull/57#issuecomment-767928447) [](ancestors = 767928447)

---
Refers to: CMakeLists.txt:4 in c6f27a6. [](commit_id = c6f27a69173408f05f536e9dc49a68f3e256f2b2, deletion_comment = False)

**Comment by danewalton** [comment]
> Thinking better about it, I'll keep version 3.2 for now.

I don't think this will jive well with the Azure SDK since we have minimum version `3.10`. If you need help updating the gate to get it `3.10` let me know.

**Comment by mamokarz** [comment]
> cmake_minimum_required(VERSION 3.2.0)

Moving to 3.10

---
In reply to: [768422933](https://github.com/Azure/azure-ulib-c/pull/57#issuecomment-768422933) [](ancestors = 768422933,767928447)

---
Refers to: CMakeLists.txt:4 in c6f27a6. [](commit_id = c6f27a69173408f05f536e9dc49a68f3e256f2b2, deletion_comment = False)

**Comment by mamokarz** [comment]
Ok, let's move to 3.10

---
In reply to: [768473207](https://github.com/Azure/azure-ulib-c/pull/57#issuecomment-768473207) [](ancestors = 768473207)

**Pull request merged**